app/main.py

```diff
@@ -1,3 +1,63 @@
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: (int, float)) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: any) -> object:
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+
+    def __iadd__(self, other: any) -> callable:
```

Comment:
The return type hint `callable` is incorrect. It should be `Distance` since the method returns the instance itself.